scripts/es2k/setup_infra.sh

```diff
@@ -0,0 +1,272 @@
+  $P4CP_INSTALL/sbin/infrap4d &
+  check_status $? "sbin/infrap4d"
+}
+
+#############################################
+##### main ##################################
+#############################################
+
+BASE_DIR="$(dirname "$(readlink -f "$0")")"
+IF_MAX=8
+MODE="host"
+REMOTE_HOST="10.10.0.2"
+
+# Displays help text
+usage() {
+  echo ""
+  echo "Usage: $0 [-i <8|16|..>] [-m <split|host>] [-r <10.10.0.2>]" 1>&2;
+  echo "Configure and setup infrastructure for deployment"
+  echo ""
+  echo "Options:"
+  echo "  -i  Num interfaces to configure for deployment"
+  echo "  -m  Mode host or split, depending on where Inframanager is configured to run"
+  echo "  -r  IP address configured by the user on the ACC-ARM complex for
+    connectivity to the Host. This is provisioned using Node Policy - comms
+    channel "[[0,3],[4,2]]". This is needed for runnning in split mode. Script will assign
+    an IP addresss from the same subnet on the Host side for connectivity."
+  echo ""
+  echo " Please set following env variables for host deployment:"
+  echo "  SDE_INSTALL - Default SDE install directory"
+  echo "  p4CP_INSTALL - Default p4cp install directory"
+  echo "  DEPEND_INSTALL - Default target dependencies directory"
+  echo ""
+  exit 1
+}
+
+
+while getopts ":i:m:r:" o; do
+  case "${o}" in
+      i)
+          i=${OPTARG}
+          ((i > 7)) || usage
+          ;;
+      m)
+          m=${OPTARG}
+          if [[ "$m" != "host" && "$m" != "split" ]]; then
+            usage
+          fi
+          ;;
+      r)
+          r=${OPTARG}
+          ;;
+      *)
+          usage
+          ;;
+  esac
+done
+
+shift $((OPTIND-1))
+
+if [ -z "${i}" ] || [ -z "${m}" ]; then
+  usage
+fi
+
+if [[ "$m" == "split" &&  -z "${r}" ]]; then
+  echo "Host-Arm connectivity IP is empty"
+  exit 1
+elif [[ "$m" == "split" ]]; then
+  if ! [[ "$r" =~ ^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
+    echo "Error: Invalid IP address format."
+    exit 1
+  fi
+fi
+
+IF_MAX=$i
+MODE=$m
+REMOTE_HOST=$r
+echo "$IF_MAX $MODE $REMOTE_HOST"
+
+if [ $MODE = "host" ]; then
+  echo "Setting up p4k8s on host"
+  check_p4cp_host_env SDE_INSTALL P4CP_INSTALL DEPEND_INSTALL
+  setup_host_dep_env
+  setup_run_env
+  install_drivers
+  sleep 3
+  copy_certs
+  create_arp_interface
+  create_pod_interfaces
+  # Run infrap4d
+  run_infrap4d
+  echo "running host"
+else
+  LAST_OCTET="${REMOTE_HOST##*.}"
+  NEW_OCTET=$((LAST_OCTET + 1))
+  NEW_IP="${REMOTE_HOST%.*}.$NEW_OCTET"
+  echo "Setting up p4k8s in split mode. Manager runs on arm"
+  install_drivers
+  sleep 3
+  ifconfig ens801f0d3 $NEW_IP/16 up
+  sleep 1
+  copy_script_to_remote
+  launch_on_remote "/usr/share/stratum/es2k/generate-certs.sh" ""
+  copy_cert_to_remote
+  SYSTEM_IP=$(get_system_ip)
+  launch_on_remote "$P4K8S_INSTALL/$ARM_SCRIPT" "$SYSTEM_IP"
+  create_arp_interface
+  create_pod_interfaces
+  echo "Remote script launched successfully!"
```

Comment:
in split mode, infrap4d doesn't run on ACC?

Reply:
It does. In that case, the script - setup_arm_infra.sh takes care of launching infrap4d and inframanager